dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/webapp/WebApp.java

```diff
@@ -333,33 +336,35 @@
     private static final String JSF_IMPL_ENABLED_PARAM = "com.ibm.ws.jsf.JSF_IMPL_ENABLED";
     private static final String JSF_IMPL_ENABLED_CUSTOM = "Custom";
     private static final String JSF_IMPL_ENABLED_NONE = "None";
-        
+
     protected DocumentRootUtils staticDocRoot = null;
     protected DocumentRootUtils jspDocRoot = null;
-        
+
     enum InitializationCollaborCommand { STARTING, STARTED, STOPPING, STOPPED };
-        
+
     protected com.ibm.wsspi.adaptable.module.Container container;
     protected MetaInfResourceFinder metaInfResourceFinder;
 
     public enum ANNOT_TYPE {
         POST_CONSTRUCT(PostConstruct.class),
         PRE_DESTROY(PreDestroy.class);
-    
+
         // This could be replaced by the annotation name.
         // See the comment on 'lookupMethod'.
         public final Class<? extends Annotation> annotationClass;
-        
+
         private ANNOT_TYPE(Class<? extends Annotation> annotationClass) {
             this.annotationClass = annotationClass;
         }
     };
-    
+
     private final AnnotatedMethods postConstructMethods = new AnnotatedMethods(ANNOT_TYPE.POST_CONSTRUCT);
     private final AnnotatedMethods preDestroyMethods = new AnnotatedMethods(ANNOT_TYPE.PRE_DESTROY);
-    
+
     private static Object[] OBJ_EMPTY = new Object[] {};
     private static Class<?>[] CLASS_EMPTY = new Class<?>[] {};
+    protected static CheckpointPhase checkpointPhase;
```

Comment:
Should not be static.  I would prefer a final var set with a constructor.  If that proves to be difficult for unit tests then just add a new constructor that takes the extra arg and have the original constructors pass null phase.